src/P2/bbmark.c

```diff
@@ -1,20 +1,43 @@
 #include <bbmark.h>
+#include <slotheap.h>
+#include <sw.h>
 
 INCLUDE_ASM("asm/nonmatchings/P2/bbmark", UpdateSwPox__FP2SWP3OXAT1UcUc);
 
-INCLUDE_ASM("asm/nonmatchings/P2/bbmark", PoxAddSw__FP2SWP3OXAT1);
+OX *PoxAddSw(SW *psw, OXA *poxa, OXA *poxaOther)
+{
+    OX *pox = (OX *)PvAllocSlotheapImpl(&psw->slotheapOx);
+    pox->pxp = (XP *)nullptr;
+    pox->psoOther = poxaOther->pso;
+    pox->poxNext = poxa->pox;
+    poxa->pox = pox;
+    return pox;
+}
 
 INCLUDE_ASM("asm/nonmatchings/P2/bbmark", PoxRemoveSw__FP2SWP3OXAT1);
 
 INCLUDE_ASM("asm/nonmatchings/P2/bbmark", PoxFromSoSo__FP2SOT0);
 
-INCLUDE_ASM("asm/nonmatchings/P2/bbmark", PxpFirstFromSoSo__FP2SOT0);
+XP *PxpFirstFromSoSo(SO *pso, SO *psoOther)
+{
+    OX *pox = PoxFromSoSo(pso, psoOther);
+    XP *pxp = (XP *)nullptr;
+    if (pox)
+    {
+        pxp = pox->pxp;
+    }
+
+    return pxp;
+}
```

Comment:
I wonder if this would be equivalent:
```cpp
XP *PxpFirstFromSoSo(SO *pso, SO *psoOther)
{
    OX *pox = PoxFromSoSo(pso, psoOther);
    return pox ? pox->pxp : (XP *)nullptr;
}
```